mnemonics/src/main/java/org/kethereum/bip32/ExtendedKey.kt

```diff
@@ -0,0 +1,257 @@
+package org.kethereum.bip32
+
+import org.kethereum.crypto.CURVE
+import org.kethereum.crypto.ECKeyPair
+import org.kethereum.crypto.Keys
+import org.kethereum.crypto.Keys.PRIVATE_KEY_SIZE
+import org.kethereum.crypto.Keys.decompressKey
+import org.kethereum.crypto.Keys.getCompressedPublicKey
+import org.kethereum.crypto.Keys.publicKeyFromPoint
+import org.kethereum.encodings.decodeBase58WithChecksum
+import org.kethereum.encodings.encodeToBase58WithChecksum
+import org.kethereum.extensions.toBytesPadded
+import org.kethereum.hashes.ripemd160
+import org.kethereum.hashes.sha256
+import org.spongycastle.jce.provider.BouncyCastleProvider
+import java.io.IOException
+import java.math.BigInteger
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.security.*
+import java.util.*
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+
+data class ExtendedKey(private val keyPair: ECKeyPair,
+                       private val chainCode: ByteArray,
+                       private val depth: Byte,
+                       private val parentFingerprint: Int,
+                       private val sequence: Int) {
+
+    init {
+        Security.addProvider(BouncyCastleProvider())
```

Comment:
Imho this is not needed as we depend on crypto

Reply:
I cleared this as well.
You are right.